torchx/CMakeLists.txt

```diff
@@ -6,39 +6,91 @@ if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
   set(CMAKE_CUDA_ARCHITECTURES 75)
 endif()
 
+if(NOT DEFINED ENV{CMAKE_TOOLCHAIN_FILE})
+  set(CMAKE_TOOLCHAIN_FILE $ENV{CMAKE_TOOLCHAIN_FILE})
+endif()
+
+if(NOT DEFINED CMAKE_BUILD_TYPE OR "${CMAKE_BUILD_TYPE}" STREQUAL "")
+    set(CMAKE_BUILD_TYPE "Release")
+endif()
+
+if(DEFINED ENV{MIX_APP_PATH})
+    set(PRIV_DIR "$ENV{MIX_APP_PATH}/priv")
+else()
+    set(PRIV_DIR "${CMAKE_CURRENT_SOURCE_DIR}/priv")
+endif()
+set(CMAKE_INSTALL_PREFIX "${PRIV_DIR}")
+if(NOT IS_DIRECTORY "${PRIV_DIR}")
+  file(MAKE_DIRECTORY "${PRIV_DIR}")
+endif()
+
+if(DEFINED ERTS_INCLUDE_DIR AND NOT "${ERTS_INCLUDE_DIR}" STREQUAL "")
+    set(ERTS_INCLUDE_DIR "${ERTS_INCLUDE_DIR}")
+else()
+    set(ERTS_INCLUDE_DIR_ONE_LINER "erl -noshell -eval \"io:format('~ts/erts-~ts/include/', [code:root_dir(), erlang:system_info(version)]), halt().\"")
+    if(WIN32)
+        execute_process(COMMAND powershell -command "${ERTS_INCLUDE_DIR_ONE_LINER}" OUTPUT_VARIABLE ERTS_INCLUDE_DIR)
+    else()
+        execute_process(COMMAND bash -c "${ERTS_INCLUDE_DIR_ONE_LINER}" OUTPUT_VARIABLE ERTS_INCLUDE_DIR)
+    endif()
+    set(ERTS_INCLUDE_DIR "${ERTS_INCLUDE_DIR}")
+endif()
```

Comment:
We can still compute this on the Elixir side and pass it as `env` in the `cmd!` function. WDYT?

Reply:
Sure! I'll copy some relevant code from `elixir_make` to do it :)